Jetchat/app/src/main/java/com/example/compose/jetchat/data/CustomChatMessage.kt

```diff
@@ -5,14 +5,17 @@ import com.aallam.openai.api.chat.ChatMessage
 import com.aallam.openai.api.chat.ChatRole
 import com.aallam.openai.api.chat.FunctionCall
 
-
-
+/**
+ * Wrapper for the final `ChatMessage` class so that we
+ * can count tokens and split the grounding from the
+ * user query for embedding-supported (RAG) requests
+ */
 class CustomChatMessage @OptIn(BetaOpenAI::class) constructor(
-    val role: ChatRole,
-    val grounding: String? = null,
-    val userContent: String? = null,
-    val name: String? = null,
-    val functionCall: FunctionCall? = null
+    public val role: ChatRole,
+    public val grounding: String? = null,
+    public val userContent: String? = null,
+    public val name: String? = null,
+    public val functionCall: FunctionCall? = null
```

Comment:
```suggestion
    val role: ChatRole,
    val grounding: String? = null,
    val userContent: String? = null,
    val name: String? = null,
    val functionCall: FunctionCall? = null
```

I think the IDE is suggesting that the public modifier is unnecessary - but if you needed them to be added feel free to disregard!